browser/android/youtube_script_injector/youtube_script_injector_tab_helper.cc

```diff
@@ -126,4 +121,46 @@ void YouTubeScriptInjectorTabHelper::PrimaryMainDocumentElementAvailable() {
   }
 }
 
+// static
+bool YouTubeScriptInjectorTabHelper::IsYouTubeVideo(const GURL& url) {
+  if (!url.is_valid() || url.is_empty()) {
+    return false;
+  }
+
+  // Check if domain is youtube.com (including subdomains).
+  if (!net::registry_controlled_domains::SameDomainOrHost(
+          url, GURL("https://www.youtube.com"),
+          net::registry_controlled_domains::INCLUDE_PRIVATE_REGISTRIES)) {
+    return false;
+  }
+
+  // Check if path is exactly "/watch" (case sensitive).
+  std::string path = url.path();
```

Comment:
* add const
* don't make std::string (a heap alloc) when we don't need it. Use std::string_view method instead:
`const auto path = url.path_piece();`
`constexpr std::string_view watch_path = "/watch";`

Reply:
Feedback addressed by [`93b7dd0` (#29431)](https://github.com/brave/brave-core/pull/29431/commits/93b7dd0b178123482b30f2738374ed81bdd6a827)